src/snyk/snykCode/code.ts

```diff
@@ -106,24 +106,33 @@ export class SnykCode implements ISnykCode {
 
       this.runAnalysis();
 
-      let result;
+      let result: FileAnalysis | null = null;
       if (this.changedFiles.size && this.remoteBundle) {
         const changedFiles = [...this.changedFiles];
         this.changedFiles.clear();
-        result = await extendAnalysis(this.remoteBundle, changedFiles, constants.MAX_PAYLOAD, this.config.source);
+        result = await extendAnalysis({ ...this.remoteBundle, files: changedFiles });
       } else {
         result = await analyzeFolders({
-          baseURL: this.config.baseURL,
-          sessionToken: this.config.snykCodeToken ?? '', // todo: handle the case appropriately
-          paths,
-          source: this.config.source,
+          connection: {
+            baseURL: this.config.baseURL,
+            sessionToken: this.config.snykCodeToken ?? '', // todo: handle the case appropriately
+            source: this.config.source,
+          },
+          analysisOptions: {
+            legacy: true,
+          },
+          fileOptions: {
+            paths,
+          }
         });
       }
 
       if (result) {
         this.remoteBundle = result;
 
-        this.analyzer.setAnalysisResults(result.analysisResults);
+        if (result.analysisResults.type == 'legacy') {
```

Comment:
Out of curiosity - why is this check needed? Is it possible that we ask for legacy analysis but the results are returned not for a legacy one?

Reply:
This is just Typescript thing. I don't know how to forcefully set that type and i also hate typecasting (with " as ")
For the second question, no, not possible

Reply:
Got it, makes sense 👍 User-defined type guards usually cover such scenarios in TS, but it's an overkill in this particular case since legacy result will be obsolete in the near future.